src/query/service/src/interpreters/access/privilege_access.rs

```diff
@@ -832,19 +859,19 @@ impl AccessChecker for PrivilegeAccess {
                     .await?;
             }
             Plan::CopyIntoTable(plan) => {
-                self.validate_access_stage(&plan.stage_table_info.stage_info, UserPrivilegeType::Read).await?;
+                self.validate_stage_access(&plan.stage_table_info.stage_info, UserPrivilegeType::Read).await?;
                 self.validate_table_access(plan.catalog_info.catalog_name(), &plan.database_name, &plan.table_name, vec![UserPrivilegeType::Insert]).await?;
                 if let Some(query) = &plan.query {
                     self.check(ctx, query).await?;
                 }
             }
             Plan::CopyIntoLocation(plan) => {
-                self.validate_access_stage(&plan.stage, UserPrivilegeType::Write).await?;
+                self.validate_stage_access(&plan.stage, UserPrivilegeType::Write).await?;
```

Comment:
what's the difference of Write And Insert?

Reply:
write contains insert. But write is unique to the stage object.

Reply:
> what's the difference of Write And Insert?

https://databend.rs/sql/sql-reference/access-control-privileges#stage-privileges

Maybe you can read this.